### PR TITLE
Added SP+TM Region RST docstrings.

### PR DESCRIPTION
**Pull request by rhyolight**

### CODE REVIEW

File: src/nupic/regions/sp_region.py [hunk 1]
```diff
@@ -478,10 +465,10 @@ def _allocateSpatialFDR(self, rfInput):
 
   def compute(self, inputs, outputs):
     """
-    Run one iteration of SPRegion's compute, profiling it if requested.
+    Run one iteration, profiling it if requested.
 
-    The guts of the compute are contained in the _compute() call so that
-    we can profile it if requested.
+    :param inputs: TODO: I need help with this. -- Matt 
+    :param outputs: TODO: I need help with this. -- Matt
```
Comment (rhyolight) [inline src/nupic/regions/sp_region.py]: @scottpurdy Can you or @mrcslws provide descriptions of theses params? The expected data structures need to be defined somewhere.

File: src/nupic/regions/tm_region.py [hunk 2]
```diff
@@ -453,10 +437,11 @@ def initialize(self):
...
   def compute(self, inputs, outputs):
     """
-    Run one iteration of TMRegion's compute, profiling it if requested.
+    Run one iteration of :class:`~nupic.regions.tm_region.TMRegion` compute, 
+    profiling it if requested.
 
-    The guts of the compute are contained in the _compute() call so that
-    we can profile it if requested.
+    :param inputs: TODO: I need help with this. -- Matt 
+    :param outputs: TODO: I need help with this. -- Matt
```
Comment (rhyolight) [inline src/nupic/regions/tm_region.py]: @scottpurdy Can you or @mrcslws provide descriptions of theses params? The expected data structures need to be defined somewhere.

**Comment by rhyolight** [comment]
Waiting for some help with a couple docstrings before ready to merge.

File: src/nupic/regions/tm_region.py (hunk 2 above)
Comment (scottpurdy) [inline src/nupic/regions/tm_region.py]:
```
:param inputs: (dict) mapping region input names to numpy.array values
:param outputs: (dict) mapping region output names to numpy.arrays that should be populated with output values by this method
```

File: src/nupic/regions/sp_region.py (hunk 1 above)
Comment (scottpurdy) [inline src/nupic/regions/sp_region.py]: same as other

**Comment by rhyolight** [comment]
Thanks @scottpurdy this is ready for review.

**Pull request merged**